plugins/processors/starlark/starlark.go

```diff
@@ -73,6 +74,17 @@ func (s *Starlark) Add(metric telegraf.Metric, acc telegraf.Accumulator) error {
 					s.Log.Errorf("Duplicate metric reference detected")
 					continue
 				}
+
+				// Previous metric was found, accept the starlark metric, add
+				// the original metric to the accumulator
+				if v.ID == origMetric.HashID() {
+					origFound = true
+					m.Accept()
+					s.results = append(s.results, origMetric)
+					acc.AddMetric(origMetric)
+					continue
+				}
```

Comment:
How about
```suggestion
				// Check if the original metric is still contained in the result set. If it was not found,
				// we drop the original metric after the loop and otherwise defer acceptance to the
				// next instance / the output.
				origFound = origFound || v.ID == origMetric.HashID()
```

Reply:
Recall we need to add the original metric to the accumulator, not the metric we got back.